aya/src/programs/socket_filter.rs

```diff
@@ -92,7 +92,10 @@ impl SocketFilter {
             .into());
         }
 
-        self.data.links.insert(SocketFilterLink { socket, prog_fd })
+        self.data.links.insert(SocketFilterLink {
+            socket,
```

Comment:
This function should be `pub fn attach<T: AsFd>` since I think this is implemented for all stdlib socket types that we could attach these programs to.